### DISCUSSION TITLE
How to 3D bounding Box  labeling and training model ?

### DISCUSSION THREAD

**Discussion by damavand1** [ultralytics/ultralytics#9213] [General]
Hi

Thanks for your perfect product for detection and segmentation

now the question is how can I label data with 3D info and train the model ?

somethings like this:
![687474703a2f2f736f726f7573686b686164656d2e636f6d2f696d672f32642d746f702d33642d626f74746f6d312e706e67 (1)](https://github.com/ultralytics/ultralytics/assets/104978921/6ae1579f-cc77-47b4-b3a8-5b0347bfea00)

for a result like this :
![3d-bbox-vid](https://github.com/ultralytics/ultralytics/assets/104978921/89e1e0c0-4337-45d7-888a-20b09a85f1be)

I found somethings like [this](https://github.com/skhadem/3D-BoundingBox), but what's state-of-the-art ?

Thank you

**Comment by pderrenger** [comment]
Hi,

Thank you so much for your kind words and your interest in 3D object detection! 🙌 Currently, Ultralytics YOLO models, including YOLOv8, are primarily designed for 2D object detection, instance segmentation, and related tasks. They don't natively support 3D bounding box labeling and training out of the box.

For 3D detection tasks, the project you found ([3D-BoundingBox](https://github.com/skhadem/3D-BoundingBox)) can be a great starting point. Another popular tool for 3D annotation is [Label Studio](https://labelstud.io/), which supports various types of data labeling, including 3D.

Since 3D object detection typically involves depth information or point clouds, you may want to explore dedicated 3D object detection frameworks or models that are designed for processing this type of data, like PointNet, Open3D, or MMDetection3D.

We appreciate your suggestion and are always looking to expand the capabilities of our models. We'll definitely keep your feedback in mind for future updates!

Happy detecting,
The Ultralytics Team 🚀

**Reply by damavand1** [reply]
Hi Pderrenger

Thanks for suggestion, I'm try MMDetection3D

Special Thanks again

**Reply by pderrenger** [reply]
@damavand1 hi Pderrenger,

You're welcome! We're glad you're giving MMDetection3D a try. If you have any questions or need assistance while exploring 3D detection, feel free to reach out. Happy coding! 😊
